### PR TITLE
Upgraded to latest version of rollup

### PR DESCRIPTION
**Pull request by usergenic**
- Added option for bundler to let rollup treeshake.
 - Added a bundler test to demonstrate treeshake with PURE annotations.

### CODE REVIEW

**Comment by usergenic** [comment]
cc @westbrook

File: packages/bundler/src/es6-rewriter.ts [hunk 1]
```diff
@@ -139,28 +138,38 @@ export class Es6Rewriter {
...
-    this._rewriteImportStatements(url, babelFile);
-    this._deduplicateImportStatements(babelFile);
-    const {code: rewrittenCode} = serialize(babelFile);
-    return {code: rewrittenCode, map: undefined};
+
+    if (output.length !== 1) {
+      throw new Error(`Failed to bundle.  Rollup generated ${
+          output.length} chunks or assets.  Expected 1.`);
+    }
+    for (const chunkOrAsset of output) {
```
Comment (justinfagnani) [inline packages/bundler/src/es6-rewriter.ts]: do you need the loop if the length is 1?

File: packages/bundler/src/es6-rewriter.ts [hunk 2]
```diff
@@ -139,28 +138,38 @@ export class Es6Rewriter {
...
+      const rolledUpDocument = await this.bundler.analyzeContents(
+          rolledUpUrl as ResolvedUrl, rolledUpCode);
+      const babelFile = assertIsJsDocument(rolledUpDocument).parsedDocument.ast;
+      this._rewriteExportStatements(url, babelFile);
+      this._rewriteImportStatements(url, babelFile);
+      this._deduplicateImportStatements(babelFile);
+      const {code: rewrittenCode} = serialize(babelFile);
+      return {code: rewrittenCode, map: undefined};
+    }
+    throw new Error(`Failed to bundle.  Rollup generated no chunks or assets.`);
```
Comment (justinfagnani) [inline packages/bundler/src/es6-rewriter.ts]: shouldn't this be redundant with the `output.length` check above?

File: packages/bundler/src/es6-rewriter.ts (hunk 1 above)
Comment (usergenic) [reply packages/bundler/src/es6-rewriter.ts]: This is a holdover from when I wasn't sure if I needed to find or merge chunks, which it doesn't look like I will.  It does obscure the intent leaving it this way.  Removed the loop.  Thanks.

File: packages/bundler/src/es6-rewriter.ts (hunk 2 above)
Comment (usergenic) [reply packages/bundler/src/es6-rewriter.ts]: Yeah but the for loop caused typescript to declare a possible undefined return.  This problem has gone away by removing the for loop and thus this throw is unneeded and now removed.

File: packages/bundler/CHANGELOG.md [hunk 3]
```diff
@@ -6,6 +6,8 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/)
 and this project adheres to [Semantic Versioning](http://semver.org/).
 
 <!-- ## Unreleased -->
+* Upgraded Rollup to version 1.3.0.
+* Added `treeshake` option to enable Rollup to remove dead code when bundling.
```
Comment (Westbrook) [inline packages/bundler/CHANGELOG.md]: 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥

File: packages/cli/src/test/integration/integration_test.ts [hunk 4]
```diff
@@ -54,8 +55,12 @@ async function serve(dirToServe: string, options: ServerOptions = {}) {
   const address = startResult.server.address();
   if (typeof address === 'string') {
     return `http://${address}`;
-  } else {
+  } else if (!isNull(address)) {
     return `http://${address.address}:${address.port}`;
+  } else {
+    // How you gonna serve without an address?  How is that even a server?  If a
+    // server can't respond to requests, is it *reall* a server?
```
Comment (Westbrook) [inline packages/cli/src/test/integration/integration_test.ts]: Sarcasm is spelled with a `y` at the end here? I mean, "reallY".
```suggestion
    // server can't respond to requests, is it *really* a server?
```

**Comment by usergenic** [comment]
@westbrook can you sign the CLA for your contribution of the Y in "really"?

**Comment by Westbrook** [comment]
That's weird. I've signed them before... in this repo. Is there a link to do it again?

Also, sorry for so much trouble over one little letter.

**Comment by usergenic** [comment]
@Westbrook I'm marking CLA: yes because that commit is immaterial and I don't know what googlebot is talking about.

**Comment by Westbrook** [comment]
Love it!

**Pull request merged**